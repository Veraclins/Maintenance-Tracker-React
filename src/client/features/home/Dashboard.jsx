import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import toastr from 'toastr';
import { PropTypes } from 'prop-types';
import history from '../../shared/utilities/history';


export class Dashboard extends Component {
  componentDidMount() {
    const {
      location, isLoggedIn,
    } = this.props;
    if (!isLoggedIn) {
      toastr.error('You must be logged in to visit your dashboard');
      return history.push('/login', { from: location.pathname });
    }
    return true;
  }

  render() {
    const { isLoggedIn, user } = this.props;
    return (
    /* eslint-disable react/no-unescaped-entities */
      <section className="body_gradient home">
        <div className="dashboard_top">
          <div className="container">
            {isLoggedIn
              ? (
                <h1 className="home_welcome">
                  Welcome back <strong>{user.firstName} {user.lastName}</strong>
                </h1>
              ) : false
            }
          </div>
        </div>
        <div className="content_section_1 section_viewer" data-section="deliver">
          <div className="container">
            <div className="layout_content wow fadeInUp" data-wow-offset="150">
              <h1>What do you want to do today?</h1>
              <span className="banner_big_title_2">DASHBOARD</span>
            </div>
          </div>
          <div className="section_1_bg" />
          <div className="how_section_bg" />
        </div>
        <div className="home_timeline_section">
          <div className="container">
            <div className="timeline_section_row">
              <div className="timeline_buttons timeline_button_left">
                <div className="wow fadeInLeft">
                  {' '}
                  <Link
                    to="/requests/create"
                    className="buttons button_default"
                  >
                    CREATE A NEW REQUEST
                  </Link>
                </div>
              </div>
              <div className="timeline_buttons timeline_button_right">
                <div className="wow fadeInRight">
                  {' '}
                  <Link
                    to="/requests"
                    className="buttons button_blank"
                  >
                    VIEW ALL REQUESTS
                  </Link>
                </div>
              </div>
              <div className="clear" />
            </div>
          </div>
          <div className="timeline_section_bg" />
          <div className="footer_top_bg" />
        </div>
      </section>
    );
  }
}

export const mapStateToProps = state => ({
  isLoggedIn: state.auth.isAuthenticated,
  location: state.router.location,
  user: state.auth.user,
});

Dashboard.propTypes = {
  isLoggedIn: PropTypes.bool.isRequired,
  user: PropTypes.shape({}).isRequired,
  location: PropTypes.shape({}).isRequired,
};

export default connect(mapStateToProps)(Dashboard);
