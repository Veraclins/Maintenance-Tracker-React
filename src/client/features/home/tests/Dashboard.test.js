import React from 'react';
import { shallow } from 'enzyme';
import toastr from 'toastr';

import { Dashboard, mapStateToProps } from '../Dashboard';

const mockFunction = jest.fn();

const user = {
  firstName: 'Clinton',
  lastName: 'Agada',
};

const history = {
  push: mockFunction,
};

const location = {
  pathname: '',
};

describe('Test the Dashboard component', () => {
  it('renders with the right view for a logged in user', () => {
    const isLoggedIn = true;
    const wrapper = shallow(<Dashboard
      isLoggedIn={isLoggedIn}
      user={user}
      location={location}
    />);
    expect(wrapper.find('section')).toHaveLength(1);
    expect(wrapper.find('h1')).toHaveLength(2);
    expect(wrapper.find('div')).toHaveLength(17);
    expect(wrapper).toMatchSnapshot();
  });
});


describe('directly invoking the componentDidMount method from component instance', () => {
  it('should call the history.push function', () => {
    const isLoggedIn = false;
    const wrapper = shallow(<Dashboard
      isLoggedIn={isLoggedIn}
      user={user}
      history={history}
      location={location}
    />);
    const instance = wrapper.instance();
    toastr.error = mockFunction;
    instance.componentDidMount();
    expect(toastr.error).toHaveBeenCalled();
  });
});

describe('Testing mapStateToProps', () => {
  it('should correctly map state  ', () => {
    const mockState = {
      auth: {
        isAuthenticated: true,
        user,
      },
      router: {
        location,
      },
    };

    const newState = mapStateToProps(mockState);

    expect(newState.isLoggedIn)
      .toEqual(true);

    expect(newState.location)
      .toEqual(location);
    expect(newState.user)
      .toEqual(user);
  });
});
