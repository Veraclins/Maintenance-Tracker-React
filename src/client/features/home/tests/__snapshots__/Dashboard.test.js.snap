// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Dashboard component renders with the right view for a logged in user 1`] = `
<section
  className="body_gradient home"
>
  <div
    className="dashboard_top"
  >
    <div
      className="container"
    >
      <h1
        className="home_welcome"
      >
        Welcome back 
        <strong>
          Clinton
           
          Agada
        </strong>
      </h1>
    </div>
  </div>
  <div
    className="content_section_1 section_viewer"
    data-section="deliver"
  >
    <div
      className="container"
    >
      <div
        className="layout_content wow fadeInUp"
        data-wow-offset="150"
      >
        <h1>
          What do you want to do today?
        </h1>
        <span
          className="banner_big_title_2"
        >
          DASHBOARD
        </span>
      </div>
    </div>
    <div
      className="section_1_bg"
    />
    <div
      className="how_section_bg"
    />
  </div>
  <div
    className="home_timeline_section"
  >
    <div
      className="container"
    >
      <div
        className="timeline_section_row"
      >
        <div
          className="timeline_buttons timeline_button_left"
        >
          <div
            className="wow fadeInLeft"
          >
             
            <Link
              className="buttons button_default"
              replace={false}
              to="/requests/create"
            >
              CREATE A NEW REQUEST
            </Link>
          </div>
        </div>
        <div
          className="timeline_buttons timeline_button_right"
        >
          <div
            className="wow fadeInRight"
          >
             
            <Link
              className="buttons button_blank"
              replace={false}
              to="/requests"
            >
              VIEW ALL REQUESTS
            </Link>
          </div>
        </div>
        <div
          className="clear"
        />
      </div>
    </div>
    <div
      className="timeline_section_bg"
    />
    <div
      className="footer_top_bg"
    />
  </div>
</section>
`;
